library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/recorder/mapper/TextWireframeMapper.kt

```diff
@@ -26,46 +26,55 @@ internal class TextWireframeMapper(
-            textPosition = view.resolveTextPosition(pixelsDensity)
+            text = resolveTextValue(view),
+            textStyle = resolveTextStyle(view, pixelsDensity),
+            textPosition = resolveTextPosition(view, pixelsDensity)
         )
     }
 
+    protected open fun resolveTextValue(textView: TextView): String {
+        return textView.text.toString()
+    }
+
     // region Internal
 
-    private fun TextView.resolveTextStyle(pixelsDensity: Float): MobileSegment.TextStyle {
+    private fun resolveTextStyle(textView: TextView, pixelsDensity: Float): MobileSegment
+    .TextStyle {
         return MobileSegment.TextStyle(
-            this.typeface.resolveFontFamily(),
-            this.textSize.toLong().densityNormalized(pixelsDensity),
-            colorAndAlphaAsStringHexa(currentTextColor, OPAQUE_AS_HEXA)
+            resolveFontFamily(textView.typeface),
+            textView.textSize.toLong().densityNormalized(pixelsDensity),
+            colorAndAlphaAsStringHexa(textView.currentTextColor, OPAQUE_AS_HEXA)
         )
     }
 
-    private fun Typeface.resolveFontFamily(): String {
+    private fun resolveFontFamily(typeface: Typeface): String {
         return when {
-            this === Typeface.SANS_SERIF -> SANS_SERIF_FAMILY_NAME
-            this === Typeface.MONOSPACE -> MONOSPACE_FAMILY_NAME
-            this === Typeface.SERIF -> SERIF_FAMILY_NAME
+            typeface === Typeface.SANS_SERIF -> SANS_SERIF_FAMILY_NAME
+            typeface === Typeface.MONOSPACE -> MONOSPACE_FAMILY_NAME
+            typeface === Typeface.SERIF -> SERIF_FAMILY_NAME
             else -> SANS_SERIF_FAMILY_NAME
         }
     }
 
-    private fun TextView.resolveTextPosition(pixelsDensity: Float): MobileSegment.TextPosition {
-        return MobileSegment.TextPosition(resolvePadding(pixelsDensity), resolveAlignment())
+    private fun resolveTextPosition(textView: TextView, pixelsDensity: Float):
+        MobileSegment.TextPosition {
+        return MobileSegment.TextPosition(
+            resolvePadding(textView, pixelsDensity),
+            resolveAlignment(textView)
+        )
     }
 
-    private fun TextView.resolvePadding(pixelsDensity: Float): MobileSegment.Padding {
+    private fun resolvePadding(textView: TextView, pixelsDensity: Float): MobileSegment.Padding {
         return MobileSegment.Padding(
-            top = this.totalPaddingTop.densityNormalized(pixelsDensity).toLong(),
-            bottom = this.totalPaddingBottom.densityNormalized(pixelsDensity).toLong(),
-            left = this.totalPaddingStart.densityNormalized(pixelsDensity).toLong(),
-            right = this.totalPaddingEnd.densityNormalized(pixelsDensity).toLong()
+            top = textView.totalPaddingTop.densityNormalized(pixelsDensity).toLong(),
+            bottom = textView.totalPaddingBottom.densityNormalized(pixelsDensity).toLong(),
+            left = textView.totalPaddingStart.densityNormalized(pixelsDensity).toLong(),
```

Comment:
won't we have any issues with RTL layout where start is on the right? How SR handles RTL btw?

Reply:
We're not there yet ;)